frontend/packages/console-dynamic-plugin-sdk/src/extensions/actions.ts

```diff
@@ -126,6 +126,8 @@ export type Action = {
   insertAfter?: string | string[];
   /** Describes the access check to perform. */
   accessReview?: AccessReviewResourceAttributes;
+  /** check if sub-catalog type is enabled or not */
+  isEnabled?: boolean;
```

Comment:
This is a generic "Action" and should not contain specific attributes for "sub-catalog". We should also just do not add actions instead of setting isEnabled=false. Just from reading the action extension it would be unclear what the difference is between "disabled" and "isEnabled". I hope we can find a solution without adding this property here.

```suggestion
```

Reply:
The existing `disabled` property will not hide the option, it will just disable it and user can see the action is disabled. So I added isEnabled to hide the option. I will check how can I achieve this without `isEnabled`.